### PR TITLE
resource/gitlab_label: Add priority attribute 

### PR DESCRIPTION
**Pull request by timofurrer**
Closes: #69

Blocked by:

* https://github.com/xanzy/go-gitlab/pull/1407

The PR above highlights the issue why this is blocked: basically go-gitlab doesn't support nullable fields in responses, which would be required for the `priority` field.

### CODE REVIEW

File: internal/provider/resource_gitlab_label_test.go [hunk 1]
```diff
@@ -52,36 +54,7 @@ func TestAccGitlabLabel_basic(t *testing.T) {
...
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckGitlabLabelExists("gitlab_label.fixme.20", &label),
-					testAccCheckGitlabLabelExists("gitlab_label.fixme.30", &label),
-					testAccCheckGitlabLabelExists("gitlab_label.fixme.40", &label),
-					testAccCheckGitlabLabelExists("gitlab_label.fixme.10", &label),
-					testAccCheckGitlabLabelAttributes(&label, &testAccGitlabLabelExpectedAttributes{
-						Name:        "FIXME11",
-						Color:       "#ff0000",
-						Description: "red label",
+						Priority:    5,
```
Comment (jtymes) [inline internal/provider/resource_gitlab_label_test.go:57]: Should there be a test validating the state is correct?

File: go.mod [hunk 2]
```diff
@@ -11,7 +11,12 @@ require (
...
 	github.com/mitchellh/hashstructure v1.1.0
 	github.com/onsi/gomega v1.18.1
-	github.com/xanzy/go-gitlab v0.57.0
+	github.com/xanzy/go-gitlab v0.58.0
 	golang.org/x/crypto v0.0.0-20220214200702-86341886e292 // indirect
+	golang.org/x/net v0.0.0-20220225172249-27dd8689420f // indirect
+	golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b // indirect
+	golang.org/x/time v0.0.0-20220224211638-0e9765cccd65 // indirect
 	google.golang.org/api v0.34.0 // indirect
+	google.golang.org/appengine v1.6.7 // indirect
```
[the remark this answers is not in the document]
Comment (timofurrer) [reply go.mod:20]: It was indirectly updated from `go-gitlab` ... I guess it's okay.

File: internal/provider/resource_gitlab_label.go [hunk 3]
```diff
@@ -43,6 +44,12 @@ var _ = registerResource("gitlab_label", func() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"priority": {
+				Description:      "The priority of the label. Must be greater or equal than zero or null to remove the priority.",
```
Comment (armsnyder) [inline internal/provider/resource_gitlab_label.go:48]: Did you test if it is possible to remove the priority once it has been set?

Comment (timofurrer) [reply internal/provider/resource_gitlab_label.go:48]: I've added a test case :)

Comment (armsnyder) [reply internal/provider/resource_gitlab_label.go:48]: The new test is comparing the value to `0`, which would be a non-null priority of value 0

**Comment by PatrickRice-KSC** [comment]
@timofurrer  - I checked out the graphql API for labels to see if that helped out with this issue... there seems to be an [API to create labels](https://docs.gitlab.com/ee/api/graphql/reference/index.html#mutationlabelcreate), and there is an API to [read labels for projects](https://docs.gitlab.com/ee/api/graphql/reference/index.html#projectlabel) along with an API to [read labels for groups](https://docs.gitlab.com/ee/api/graphql/reference/index.html#grouplabel) however there appears to be no mutation for actually updating a label. So no love here for helping with this issue :-/

**Comment by timofurrer** [comment]
> Hey @timofurrer - I wonder if we could apply the same workaround for go-gitlab here that we did over here: [#971 (comment)](https://github.com/gitlabhq/terraform-provider-gitlab/pull/971#discussion_r835801777)

@PatrickRice-KSC hmmm, where would you see the point to hook into the go-gitlab response handling? I guess what we could do is just query the API ourselves with a custom response struct ...

**Comment by PatrickRice-KSC** [comment]
@timofurrer  - Oh shoot, my bad. It's in the _response_, not the _request_. I misread the note. Seems like we'd have to make a custom call at this point...

**Pull request closed without merging**